### PR TITLE
Upgrade 7zip

### PR DESCRIPTION
**Pull request by cocreature**
changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Comment by cocreature** [comment]
fwiw this isn’t necessary for anything but I made the PR to test if it fixes an unrelated CI issue and now that I have it it seems useful to upgrade anyway.

File: dev-env/windows/manifests/7zip.json [hunk 1]
```diff
@@ -1,21 +1,41 @@
...
-  "license": {
-    "identifier": "LGPL-2.0-only,BSD-3-Clause",
-    "url": "https://www.7-zip.org/license.txt"
-  },
-  "version": "18.05",
-  "architecture": {
-    "64bit": {
-      "url": "https://www.7-zip.org/a/7z1805-x64.msi",
-      "hash": "898c1ca0015183fe2ba7d55cacf0a1dea35e873bf3f8090f362a6288c6ef08d7"
+    "homepage": "https://www.7-zip.org/",
```
Comment (SamirTalwar-DA) [inline dev-env/windows/manifests/7zip.json:2]: We use 2-space indentation for all the files in this directory. Might be worth staying consistent.

File: dev-env/windows/manifests/7zip.json [hunk 2]
```diff
@@ -1,21 +1,41 @@
...
+            "hash": "b49d55a52bc0eab14947c8982c413d9be141c337da1368a24aa0484cbb5e89cd"
+        }
+    },
+    "extract_dir": "Files/7-Zip",
+    "bin": "7z.exe",
+    "checkver": {
+        "url": "https://www.7-zip.org/download.html",
+        "regex": "Download 7-Zip ([\\d.]+)"
+    },
+    "autoupdate": {
```
Comment (SamirTalwar-DA) [inline dev-env/windows/manifests/7zip.json:25]: This is pretty cool. 😁

File: dev-env/windows/manifests/7zip.json (hunk 1 above)
Comment (cocreature) [reply dev-env/windows/manifests/7zip.json:2]: I copied this from scoop upstream, I’d rather not make unnecessary modifications.

File: dev-env/windows/manifests/7zip.json (hunk 2 above)
Comment (cocreature) [reply dev-env/windows/manifests/7zip.json:25]: I don’t expect to actually use this but it shouldn’t cause problems and I like staying as close to upstream as possible :slightly_smiling_face:

**Pull request merged**